nucypher/policy/conditions/lingo.py

```diff
@@ -104,21 +106,44 @@ def __init__(self, comparator: str, value: Any):
 
         self.comparator = comparator
         self.value = value
+        self.key = key
 
     def _sanitize_value(self, value):
         try:
             return ast.literal_eval(str(value))
         except Exception:
             raise self.InvalidExpression(f'"{value}" is not a permitted value.')
 
+    def _process_data(self, data: Any) -> Any:
+        """
+        If a key is specified, return the value at that key in the data if data is a dict or list-like.
+        Otherwise, return the data.
+        """
+        if self.key and isinstance(data, dict):
+            try:
+                processed_data = data[self.key]
+            except KeyError:
+                raise KeyError(f"Key '{self.key}' not found in return data.")
```

Comment:
Since this method is executed at condition evaluation time, I think this should raise an exception caught via `_utils.evaluate_condition(...)`, eg. something like `InvalidCondition`. Same for the `IndexError` on L131.

Reply:
I'd also like to see a higher-order exception raised here to follow the emerging convention.  

> The condition isn't necessarily invalid, they key just isn't present.

Interesting take - IMO if the return value of an on-chain funtion returns a different type than what is expected by the condition, the condition is invalid, malformed, or at least incorrect.  Perhaps you can introduce a new type if exception class here: `nucypher.policy.conditions.exceptions.ReturnValueTypeError` or similar.  This can potentially also encompass other type-related problems mentioned in my prior comments.

Reply:
Don't get me wrong, I wasn't forcing an exception on you, my point was that we shouldn't raise a `KeyError`. If you raise a `KeyError` here, the `lingo.eval()` raises a `KeyError` which then probably gets turned into an HTTP 500 generalized internal error? (I think it does). If so, this is not really helpful.

The question becomes is a key not found in the data, 1) a condition failure or 2) an invalid condition. In your case you are considering it a failure, for which we can do better than just a generalized 500. It can be a separate exception (in exception.py) but I don't think it should be a generic `KeyError`, but rather something more specific and therefore informative. Same for `IndexError`. This exception would then be caught and handled appropriately (message and status code) in `evaluate_conditions(...)` in `_utils.py`.